src/app/(main)/dashboard/components/Calendar/Calendar.logic.ts

```diff
@@ -12,16 +12,25 @@ import {
   addMonths,
   startOfDay,
   endOfDay,
+  parseISO,
 } from "date-fns";
 import { useState } from "react";
-import { SprintData } from "@/app/(main)/dashboard/mocks/voyageDashboardData";
-
-export const useCalendarLogic = (sprintData?: SprintData) => {
+import { EventList } from "@/app/(main)/dashboard/components/voyage-dashboard/getDashboardData";
+import { Sprint } from "@/store/features/sprint/sprintSlice";
+
+export const useCalendarLogic = (
+  sprintsData?: Sprint[],
+  currentSprintNumber?: number | null,
+  meetingsData?: EventList[],
+) => {
   const currentDate = new Date();
```

Comment:
This is the date I mentioned in the meeting. We should move this to redux. I'll let you know tomorrow which slice it should be put in.

Reply:
I think user slice would be good place to put this

Reply:
I'm sorry, but I'm not entirely clear on what date you mean. Are you suggesting something similar to the example below? Save the _new Date_ on the user state? Or we want to set it with a reducer on the slice?

```
export interface User {
  id: string;
  firstName: string;
  lastName: string;
  countryCode: string;
  discordId: string;
  githubId: string;
  twitterId: string;
  linkedinId: string;
  email: string;
  timezone: string;
  avatar: string;
  voyageTeamMembers: VoyageTeamMember[];
  currentDate: Date;
}

const initialState: User = {
  id: "",
  firstName: "",
  lastName: "",
  countryCode: "",
  discordId: "",
  githubId: "",
  twitterId: "",
  linkedinId: "",
  email: "",
  timezone: "",
  avatar: "",
  voyageTeamMembers: [],
  currentDate: new Date(),
};
```

Reply:
So, I'm not really sure off top my head what the difference is. What I had in mind was, when user logs in we can grab the user's date and then dispatch that to redux store. So that would be in the auth provider component.

Reply:
Something like that?

```
useEffect(() => {
  if (user) {
    dispatch(clientSignIn());
    // Add the currentDate field to the user object
    const userWithDate = { ...user, currentDate: new Date() };
    // Dispatch the getUserState action with the user object
    dispatch(getUserState(userWithDate));
  }

  if (error) {
    dispatch(clientSignOut());
  }
}, [dispatch, user, error]);
```

Reply:
Yea, something like that